src/main/jbake/content/community/docs/developer-guide/alerts.adoc

```diff
@@ -389,3 +451,420 @@ public class Action {
+
+== Elasticsearch Integration
+
+=== Elasticsearch Alerter
+
+The Elasticsearch Alerter will listen for triggers tagged with "Elasticsearch" tag. The Alerter will schedule a
+periodic query to an Elasticsearch system with the info provided from the tagged trigger context. The Alerter
+will convert Elasticsearch documents into Hawkular Alerting Events and send them into the Alerting engine.
+
+The Elasticsearch Alerter uses the following conventions for trigger tags and context:
+
+[cols="^2,10"]
+|=======================
+| *Required* |
+_trigger.tags["Elasticsearch"] = "<any description>"_ +
+ +
+An _Elasticsearch_ tag is required for the alerter to detect this trigger will query to an Elasticsearch system. +
+Value is not necessary, it can be used as a description, it is reserved for future uses. +
+ +
+i.e.
+ +
+trigger.tags["Elasticsearch"] = ""                          // Empty value is valid +
+trigger.tags["Elasticsearch"] = "OpenShift Logging System"  // It can be used as description
+| *Required* |
+_trigger.context["timestamp"] = "<timestamp field>"_ +
+ +
+Documents fetched from Elasticsearch need a date field to indicate the timestamp. +
+This timestamp will be used in the queries to fetch documents in interval basis. +
+ +
+If there is not defined a specific pattern under the trigger.context["timestamp.pattern"] it will follow the default patterns: +
+ +
+"yyyy-MM-dd'T'HH:mm:ss.SSSSSSZ" +
+"yyyy-MM-dd'T'HH:mm:ssZ"
+| *Required* |
+_trigger.context["mapping"] = "<mapping_expression>"_ +
+ +
+A mapping expression defines how to convert an Elasticsearch document into a Hawkular Event. +
+ +
+Mapping expression syntax (BNF): +
+ +
+<mapping_expression> ::= <mapping> \| <mapping> "," <mapping_expression> +
+<mapping> ::= <elasticsearch_field> [ "\|" "'" <DEFAULT_VALUE> "'" ] ":" <hawkular_event_field>
+<elasticsearch_field> ::= "index" \| "id" \| <SOURCE_FIELD> +
+<hawkular_event_field> ::= "id" \| "ctime" \| "dataSource" \| "dataId" \| "category" \| "text" \| "context" \| "tags" +
+ +
+A minimum mapping for the "dataId" is required. +
+If a mapping is not present in an Elasticsearch document it will return an empty value. +
+It is possible to define a default value for cases when the Elasticsearch field is not present. +
+Special Elasticsearch metafields "_index" and "_id" are supported under "index" and "id" labels. +
+ +
+i.e.
+ +
+trigger.context["mapping"] = "level\|'INFO':category,@timestamp:ctime,message:text,hostname:dataId,index:tags" +
+| *Optional* |
+_trigger.context["interval"] = "[0-9]+[smh]"_ +
+ +
+Defines the periodic interval when a query will be performed against an Elasticsearch system. +
+If not value provided, default one is "2m" (two minutes). +
+ +
+i.e.
+ +
+trigger.context["interval"] = "30s" +
+will perform queries each 30 seconds fetching new documents generated on the last 30 seconds, using the timestamp field provided in the Alerter tag.
```

Comment:
performs a new query 30 seconds after completing the previous query, fetching documents generated since that time, using the timestamp field provided in _trigger.context["timestamp"]_.